docs/dev/infrastructure.rst

```diff
@@ -1,86 +1,201 @@
-
-The selection tool allows the user to specify a specific range on the graph. This is defined by the underlying Glue Library upon which jdaviz relies on as a "Glue Subset." Thus throughout this section, we will refer to these user defined ranges as "subsets." Effectively, the selection tool defines a mask that can be thought of as "definition" of which data is and is not included in the subset. Upon extraction via `specviz.get_spectral_regions()`, the method will return a new `specutils.SpectralRegion` object that applies that mask atop of the proper region (data) displayed, and realizes the subset the user defined in jdaviz. 
-
-Related Documents
-=================
+* *Spectrum viewer*: A widget that shows a 1D astronomical data set, primarily aimed at
+  astronomical spectra. Interactivity includes panning, zooming, and region marking.
+* *Image viewer*: A widget that shows an astronomical image, along with its on-sky
+  coordinates when WCS are available. Interactivity includes panning, zooming, stretch
+  (contrast and scale), and cut values.
+* *Cube slicer*: A widget for displaying slices or similar aggregate operations on
+  spectroscopic data cubes. While similar to the image viewer in appearence and
+  interactive capabilities, the core difference is that the main data object is
+  expected to be a data cube rather than a 2D image, and this is reflected in additional
+  aggregation/slicing operations.
+* *Table viewer*: A widget to show tabular datasets like `astropy.table.Table` objects.
+  Primarily meant to be combined with other viewers to examine the complete set of
+  properties from a selection made in another viewer. Interactivty focuses on sorting
+  and selection of specific rows (to then be highlighted in other viewers or interacted
+  with in notebook/lab).
+
+In addition to the componenent widgets above, there are also *plugins* that goes with
+them to provide the necessary user interactions. Each plugin is specialized to do one
+thing, e.g., a "model fitting" plugin to allow users to fit
+:ref:`astropy models <astropy:astropy-modeling>` to spectra.
+
+Data Analysis and I/O Libraries
+===============================
+
+The above layers are focused primarily on visualization. All actual *operations* and
+analysis tasks to be applied to visualized astronomical dataset are to be implemented
+in the respective Python libraries. It is important to note that these libraries are
+*independent* efforts from Jdaviz, and can therefore be used in whole, part, or not
+at all with the Jdaviz tools. This allows a full range of workflows, while also
+maintaining transparency to scientific users in regards to exactly how an operation
+in the Jdaviz tools actually works; i.e., they can at any time use the library directly
+instead of accessing it through Jdaviz.
+
+Some common libraries include (this list is not exhaustive):
+
+* astropy_ (general astronomy-related functionality)
```

Comment:
Does `astropy` need a link to its documentation as well? There's a trailing underscore as is.

Reply:
I don't think so, the rendered HTML links it fine. It must have been defined here already:

https://github.com/astropy/sphinx-astropy/blob/95cbe0997cd0b6e91f22ccae07b8dd265c86247f/sphinx_astropy/conf/v1.py#L96